### PR TITLE
Implement a BibTeX log parser

### LINKED ISSUE #2324: BiBTeX log (.blg) files should be parsed for warnings and shown as problems
**Is your feature request related to a problem? Please describe.**
I have the following lines in my BiBTeX log file, without noticing it at all:
```
Warning--string name "ccr" is undefined
--line 35 of file paper.bib
Warning--string name "ccr" is undefined
--line 125 of file paper.bib
...
Warning--empty journal in Author2013
Warning--empty journal in Bthor2008
```

**Describe the solution you'd like**
These should be listed as problems.

### PR DESCRIPTION
**Pull request by jlelong**
Close #2324.

### CODE REVIEW

**Comment by jlelong** [comment]
I would be happy to add more regex to catch bibtex warnings or errors if anyone has examples of such logs.

File: src/components/parser/biblogparser.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            const filename = this.resolveAuxFile(result[2], rootFile)
+            this.buildLog.push({ type: 'error', file: filename, text: result[1], line: 1 })
+        }
+
+        this.extension.logger.addLogMessage(`BibTeX log parsed with ${this.buildLog.length} messages.`)
+        this.extension.compilerLogParser.showCompilerDiagnostics(this.compilerDiagnostics, this.buildLog, 'BibTeX')
+    }
+
+    private resolveAuxFile(filename: string, rootFile: string): string {
+        filename = filename.replace('.aux', '.tex')
```
Comment (tamuratak) [inline src/components/parser/biblogparser.ts:64]: Use `/\.aux$/` to ensure it matches the extension.

File: src/components/manager.ts [hunk 2]
```diff
@@ -342,14 +342,44 @@ export class Manager {
         return undefined
     }
 
+        /**
```
Comment (tamuratak) [inline src/components/manager.ts:345]: Remove extra spaces.

File: src/components/parser/compilerlog.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+export interface LogEntry { type: string, file: string, text: string, line: number }
+
+export class CompilerLogParser {
+    // private readonly extension: Extension
+    private readonly latexLogParser: LatexLogParser
+    private readonly bibLogParser: BibLogParser
+    isLaTeXmkSkipped: boolean = false
+
+    constructor(extension: Extension) {
+        // this.extension = extension
```
Comment (tamuratak) [inline src/components/parser/compilerlog.ts:38]: You can remove the comment-outed line.

**Review by tamuratak — CHANGES_REQUESTED** [review]

**Comment by jlelong** [comment]
One more issue. When `latexmk` runs `bibtex` twice, the problems are duplicated. Need to trim logs.

**Review by tamuratak — APPROVED** [review]

**Pull request merged**